proof/courier.go

```diff
@@ -1154,10 +1155,17 @@ func (h *HashMailCourier) SetSubscribers(
 var _ Courier = (*HashMailCourier)(nil)
 
 // UniverseRpcCourierCfg is the config for the universe RPC proof courier.
+//
+// nolint:lll
 type UniverseRpcCourierCfg struct {
 	// BackoffCfg configures the behaviour of the proof delivery
 	// functionality.
 	BackoffCfg *BackoffCfg
+
+	// ServiceRequestTimeout defines the maximum duration we'll wait for
+	// a courier service to handle our outgoing request during a connection
+	// attempt, or when delivering or retrieving a proof.
+	ServiceRequestTimeout time.Duration `long:"servicerequestimeout" description:"The maximum duration we'll wait for a courier service to handle our outgoing request during a connection attempt, or when delivering or retrieving a proof."`
```

Comment:
Shouldn't the hashmail courier / every courier implementation have a similar timeout? And then use it on initialization.

For hashmail I think that would be in `NewHashMailBox`, you could add the same child `ctx` object as you're doing for the Universe courier.

Reply:
Yes you're right! Nice catch. Hahsmail is blocking also now, so would be good to have that timeout.